client/app/components/DialogWrapper.jsx

```diff
@@ -0,0 +1,206 @@
+import { isFunction } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+
+/**
+  Wrapper for dialogs based on Ant's <Modal> component.
+
+
+  Using wrapped dialogs
+  =====================
+
+  Wrapped component is an object with two fields:
+
+    {
+      showModal: (dialogProps) => object({
+          result: Promise,
+          close: (result) => void,
+          dismiss: (reason) => void,
+        }),
+      Component: React.Component, // wrapped dialog component
+    }
+
+  To open dialog, use `showModal` method; optionally you can pass additional properties that
+  will be expanded on wrapped component:
+
+    const dialog = SomeWrappedDialog.showModal()
+
+    const dialog = SomeWrappedDialog.showModal({ greeting: 'Hello' })
+
+  To get result of modal, use `result` property:
+
+    dialog.result
+      .then(...) // pressed OK button or used `close` method; resolved value is a result of dialog
+      .catch(...) // pressed Cancel button or used `dismiss` method; optional argument is a rejection reason.
+
+  Also, dialog has `close` and `dismiss` methods that allows to close dialog by caller. Passed arguments
+  will be used to resolve/reject `dialog.result` promise. `update` methods allows to pass new properties
+  to dialog.
+
+
+  Creating a dialog
+  ================
+
+  1. Add imports:
+
+    import { wrap as wrapDialog, DialogPropType } from 'path/to/DialogWrapper';
+
+  2. define a `dialog` property on your component:
+
+    propTypes = {
+      dialog: DialogPropType.isRequired,
+    };
+
+  `dialog` property is an object:
+
+    {
+      props: object, // properties for <Modal> component;
+      close: (result) => void, // method to confirm dialog; `result` will be returned to caller
+      dismiss: (reason) => void, // method to reject dialog; `reason` will be returned to caller
+    }
+
+  3. expand additional properties on <Modal> component:
+
+    render() {
+      const { dialog } = this.props;
+      return (
+        <Modal {...dialog.props}>
+      );
+    }
+
+  4. wrap your component and export it:
+
+    export default wrapDialog(YourComponent).
+
+  Your component is ready to use. Wrapper will manage <Modal>'s visibility and events.
+  If you want to override behavior of `onOk`/`onCancel` - don't forget to close dialog:
+
+    customOkHandler() {
+      this.saveData().then(() => {
+         this.props.dialog.close({ success: true }); // or dismiss();
+      });
+    }
+
+    render() {
+      const { dialog } = this.props;
+        return (
+          <Modal {...dialog.props} onOk={() => this.customOkHandler()}>
+        );
+    }
+
+
+  Settings
+  ========
+
+  You can setup this wrapper to use custom `Promise` library (for example, Bluebird):
```

Comment:
@kravets-levko can you add the reasoning here?

Reply:
Just because it was simple fix for unhandled rejections (I realized that we don't see them from `$uibModal` because it uses `$q` service (AngularJS promises) - then this idea came itself). Also I saw this approach in some (couldn't remember in which :sweat_smile:) libraries that allow to override some default dependency.

Reply:
I meant add the reasoning for using a custom Promise in the code :)

```suggestion
  To avoid "unhandledrejection" exception that would fire with native Promises, you can setup this wrapper to use custom `Promise` library (for example, Bluebird):
```

Reply:
Super :+1: Just changed text to be more generic :wink: